apache/gravitino
# [#4718] fix(iceberg):  use unified logic to transform catalog backend name to handle the renaming of catalog

### What changes were proposed in this pull request?

1. Spark,Trino, Iceberg catalog and Iceberg REST server use  `getCatalogBackendName` to get catalog backend name.
2. change the default backend name to catalog backend, like `jdbc`, it will not change after rename.

### Why are the changes needed?

Fix: #4718 

### Does this PR introduce _any_ user-facing change?
Yes

### How was this patch tested?
1. create a jdbc catalog with catalog backend name, check whether can see the schema after rename
1. create a jdbc catalog without catalog backend name, check whether can see the schema after rename
